File: content/snippets/python/language_basics_hashing_and_checksums.py
#!/usr/bin/env python3
import hashlib
import zlib


def main() -> None:
    text = "hello-world"
    encoded = text.encode("utf-8")

    sha256_hex = hashlib.sha256(encoded).hexdigest()
    crc32_value = zlib.crc32(encoded)

    print("text:", text)
    print("sha256:", sha256_hex)
    print("crc32:", crc32_value)


if __name__ == "__main__":
    main()

File: content/snippets/typescript/language_basics_hashing_and_checksums.ts
#!/usr/bin/env ts-node

import { createHash } from "node:crypto";
import { setImmediate } from "node:timers";

function checksum(text: string): number {
  return Array.from(text).reduce((acc, ch) => acc + ch.charCodeAt(0), 0);
}

function main(): void {
  const text = "hello-world";
  const sha256Hex = createHash("sha256").update(text).digest("hex");
  const sum = checksum(text);

  console.log("text:", text);
  console.log("sha256:", sha256Hex);
  console.log("checksum:", sum);
}

main();
setImmediate(() => {});
